From 439fba408d9c40a0d63b5fb50c17708abc9c002d Mon Sep 17 00:00:00 2001
From: rikardo <rikard.olander@hostmobility.com>
Date: Tue, 4 Feb 2025 08:53:12 +0000
Subject: [PATCH 1/1] Use dev board dtsi for hmm with modifications
V.1. Remove unused parts and add i2c buss handling which is the most important part. Disable eth2 that is not supported by hmm board.
Enable otg usb.

V.2 Fixup eeprom, usb cc and led controller i2c address.
    Add gpio leds so led 3 and 4 are lit.
    fix usb-c to operate as host for usb memory flash.

---
 arch/arm/dts/k3-am62-verdin-dev.dtsi          | 128 ++++++------------
 arch/arm/dts/k3-am62-verdin.dtsi              |   5 +
 .../dts/k3-am625-verdin-wifi-dev-u-boot.dtsi  |  86 ++++++++++--
 3 files changed, 119 insertions(+), 100 deletions(-)

diff --git a/arch/arm/dts/k3-am62-verdin-dev.dtsi b/arch/arm/dts/k3-am62-verdin-dev.dtsi
index 54dabe14371..738942fa942 100644
--- a/arch/arm/dts/k3-am62-verdin-dev.dtsi
+++ b/arch/arm/dts/k3-am62-verdin-dev.dtsi
@@ -11,23 +11,13 @@
 /* Verdin ETHs */
 &cpsw3g {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_rgmii1>, <&pinctrl_rgmii2>;
+	pinctrl-0 = <&pinctrl_rgmii1>;
 	status = "okay";
 };
 
 /* MDIO, shared by Verdin ETH_1 (On-module PHY) and Verdin ETH_2_RGMII */
 &cpsw3g_mdio {
 	status = "okay";
-
-	cpsw3g_phy1: ethernet-phy@7 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <7>;
-		interrupt-parent = <&main_gpio0>;
-		interrupts = <38 IRQ_TYPE_EDGE_FALLING>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_eth2_rgmii_int>;
-		micrel,led-mode = <0>;
-	};
 };
 
 /* Verdin ETH_1 (On-module PHY) */
@@ -35,73 +25,56 @@
 	status = "okay";
 };
 
-/* Verdin ETH_2_RGMII */
-&cpsw_port2 {
-	phy-handle = <&cpsw3g_phy1>;
-	phy-mode = "rgmii-rxid";
-	status = "okay";
-};
-
-/* Verdin PWM_1, PWM_2 */
-&epwm0 {
-	status = "okay";
-};
-
-/* Verdin PWM_3_DSI */
-&epwm1 {
-	status = "okay";
-};
-
-&main_gpio0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ctrl_sleep_moci>,
-		    <&pinctrl_gpio_5>,
-		    <&pinctrl_gpio_6>,
-		    <&pinctrl_gpio_7>,
-		    <&pinctrl_gpio_8>;
-};
-
 /* Verdin I2C_1 */
 &main_i2c1 {
 	status = "okay";
 
-	/* IO Expander */
-	gpio_expander_21: gpio@21 {
-		compatible = "nxp,pcal6416";
-		reg = <0x21>;
-		#gpio-cells = <2>;
+	max7321: gpio-expander@68 {
+		compatible = "maxim,max7320";
+		reg = <0x68>; 
 		gpio-controller;
+		#gpio-cells = <2>;
+		gpio-line-names =
+				"green_led_2", "red_led_2",
+				"green_led_1", "red_led_1",
+				"green_led_4", "red_led_4",
+				"green_led_3", "red_led_3";
+		interrupt-controller;
+		#interrupt-cells = <2>;
 	};
 
-	/* Current measurement into module VCC */
-	hwmon@40 {
-		compatible = "ti,ina219";
-		reg = <0x40>;
-		shunt-resistor = <10000>;
+	/*USB-C controller*/
+	typec@1d {
+		compatible = "nxp,ptn5150";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usb1_irq>;
+		interrupt-parent = <&main_gpio0>;
+		interrupts = <38 IRQ_TYPE_LEVEL_HIGH>;
+		reg = <0x1d>;
+		status ="okay";
 	};
 
-	/* Temperature sensor */
-	sensor@4f {
-		compatible = "ti,tmp75c";
-		reg = <0x4f>;
+	eeprom@54 {
+		compatible = "st,24c02", "atmel,24c04";
+		reg = <0x54>;
+		status = "okay";
+		//vcc-supply = <&reg_base_3v3>;
 	};
 
-	/* EEPROM */
-	eeprom@57 {
-		compatible = "st,24c02", "atmel,24c02";
-		reg = <0x57>;
-		pagesize = <16>;
+	/* HW confirmed, TMP103D has address 0x73 */
+	temperature-sensor@73 {
+		compatible = "ti,tmp103";
+		reg = <0x73>;
+		status = "disable";
 	};
-};
-
-/* Verdin I2C_2_DSI */
-&main_i2c2 {
-	status = "okay";
-};
-
-/* Verdin I2C_4_CSI */
-&main_i2c3 {
-	status = "okay";
+	/*
+	Add HSM module
+	hsm@48 {
+		compatible = "There is no driver";
+		reg = <0x48>;
+		status = "okay";
+	};
+	*/
 };
 
 /* Verdin CAN_1 */
@@ -133,29 +106,11 @@
 };
 #endif
 
-&mcu_gpio0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_gpio_1>,
-		    <&pinctrl_gpio_2>,
-		    <&pinctrl_gpio_3>,
-		    <&pinctrl_gpio_4>;
-};
-
-/* Verdin I2C_3_HDMI */
-&mcu_i2c0 {
-	status = "okay";
-};
-
 /* Verdin UART_4 */
 &mcu_uart0 {
 	status = "okay";
 };
 
-/* Verdin QSPI_1 */
-&ospi0 {
-	status = "okay";
-};
-
 /* Verdin SD_1 */
 &sdhci1 {
 	ti,driver-strength-ohm = <33>;
@@ -169,15 +124,16 @@
 
 &usb0 {
 	status = "okay";
+	dr_mode = "host";
 };
 
 /* Verdin USB_2 */
 &usbss1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &usb1 {
-	status = "okay";
+	status = "disabled";
 };
 
 /* Verdin CTRL_WAKE1_MICO# */
diff --git a/arch/arm/dts/k3-am62-verdin.dtsi b/arch/arm/dts/k3-am62-verdin.dtsi
index 912297d01f4..ef4b159e4d9 100644
--- a/arch/arm/dts/k3-am62-verdin.dtsi
+++ b/arch/arm/dts/k3-am62-verdin.dtsi
@@ -688,6 +688,11 @@
 			AM62X_IOPAD(0x0060, PIN_OUTPUT, 1) /*  (R25) GPMC0_AD9.VOUT0_DATA17 */
 		>;
 	};
+	pinctrl_usb1_irq: usb1-irq-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x009c, PIN_INPUT, 7) /* (V25) GPMC0_WAIT1.GPIO0_38 */ /* GPIO_SODIMM_189 */
+		>;
+	};
 };
 
 &mcu_pmx0 {
diff --git a/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi b/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
index 487485c8959..4549704b02a 100644
--- a/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
+++ b/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
@@ -8,7 +8,7 @@
 / {
 	aliases {
 		eeprom0 = &eeprom_module;
-		eeprom1 = &eeprom_carrier_board;
+		eeprom1 = &eeprom_hmm_board;
 		eeprom2 = &eeprom_display_adapter;
 	};
 
@@ -19,6 +19,71 @@
 	memory@80000000 {
 		bootph-pre-ram;
 	};
+	/*	The leds are read from the highest to lowest position on the module when the PCB is in horizontal position.
+		Default: leds lit on the two lowest green/red LED (3 and 4) and the rest are off.
+	*/
+	leds {
+		compatible = "gpio-leds";
+
+		green_led_1 {
+			label = "green:led_1";
+			function = "led_1";
+			gpios = <&max7321 2 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_1 {
+			label = "red:led_1";
+			function = "led_1";
+			gpios = <&max7321 3 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_2 {
+			label = "green:led_2";
+			function = "led_2";
+			gpios = <&max7321 0 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_2 {
+			label = "red:led_2";
+			function = "led_2";
+			gpios = <&max7321 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_3 {
+			label = "green:led_3";
+			function = "led_3";
+			gpios = <&max7321 6 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		red_led_3 {
+			label = "red:led_3";
+			function = "led_3";
+			gpios = <&max7321 7 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		green_led_4 {
+			label = "green:led_4";
+			function = "led_4";
+			gpios = <&max7321 4 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		red_led_4 {
+			label = "red:led_4";
+			function = "led_4";
+			gpios = <&max7321 5 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+
+	};
+
 };
 
 &cbass_main {
@@ -69,9 +134,6 @@
 	bootph-pre-ram;
 };
 
-&cpsw3g_phy1 {
-	bootph-pre-ram;
-};
 
 &cpsw_port1 {
 	bootph-pre-ram;
@@ -162,12 +224,9 @@
 		reg = <0x50>;
 		pagesize = <16>;
 	};
-
-	/* EEPROM on carrier board */
-	eeprom_carrier_board: eeprom@57 {
+	eeprom_hmm_board: eeprom@54 {
 		compatible = "i2c-eeprom";
-		reg = <0x57>;
-		pagesize = <16>;
+		reg = <0x54>;
 	};
 };
 
@@ -244,14 +303,13 @@
 };
 
 &usbss0 {
-	bootph-pre-ram;
+	status = "okay";
 };
 
 &usb0 {
-	dr_mode = "peripheral";
-	/* Since role switching is not supported in U-Boot */
-	/delete-property/ extcon;
-	bootph-pre-ram;
+	dr_mode = "host";
+	dr_mode = "host";
+	status = "okay";
 };
 
 &verdin_ctrl_sleep_moci {
-- 
2.34.1

